src/test.rs

```diff
@@ -551,3 +551,31 @@ fn test_join_lengths() {
     );
     assert_eq!(CUSTOM_JOIN_CALLS.load(Ordering::SeqCst), 1);
 }
+
+#[test]
+fn test_standard_hasher() {
+    fn hash<H>(mut hasher: H) -> u64
+    where
+        H: core::hash::Hasher,
+    {
+        hasher.write_u32(2020);
+        hasher.write_u8(7);
+        hasher.write_u8(23);
+        hasher.write(b"Hello, World!");
+
+        hasher.finish()
+    }
+
+    let hash = hash(crate::Hasher::new());
+    let expected = {
+        let mut hash = crate::Hasher::new();
+        hash.update(&2020u32.to_ne_bytes());
+        hash.update(&[7]);
+        hash.update(&[23]);
+        hash.update(&b"Hello, World!"[..]);
```

Comment:
This is a good test, but I think it also highlights two of the reasons we've previously decided not to implement `Hasher`:

1. The `std::hash::Hash` serialization of standard types differs from machine to machine (`to_ne_bytes`).
2. I don't think there's any guarantee that the serialization of a type is stable over time. The standard library is within its rights to break this test in a future release.